native/reader.h

```diff
@@ -216,6 +216,127 @@ class TermDictionary {
     bool is_mutable;
 };
 
+class OntologyCodeDictionary {
```

Comment:
Why do we need a new class for this?

Reply:
TermDictionary maps from string to uint32. For code descriptions, I need string to string mapping.

Reply:
You should make this a C++ template so that you can have the same code for string and uint32.